src/test/java/com/saralein/server/controller/form/FormControllerTest.java

```diff
@@ -0,0 +1,56 @@
+package com.saralein.server.controller.form;
+
+import com.saralein.server.Controller.form.FormController;
+import com.saralein.server.data.FormStore;
+import com.saralein.server.request.Request;
+import com.saralein.server.request.RequestParser;
+import com.saralein.server.response.Header;
+import com.saralein.server.response.Response;
+import java.util.HashMap;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class FormControllerTest {
+    private byte[] bodyArray;
+    private FormController formController;
+    private Response formResponse;
+    private FormStore formStore;
+
+    @Before
+    public void setUp() {
+        String body = "<h3>My: Data</h3><h3>More: Stuff</h3>";
+
+        bodyArray = body.getBytes();
+
+        RequestParser requestParser = new RequestParser();
+        Request request = requestParser.parse("POST /form HTTP/1.1\r\n\r\nBody: My=Data&More=Stuff");
```

Comment:
Feels like a lot of tests are tightly coupled to your requestParser implementation, is there a more expressive way to build a request?

Reply:
Maybe I will look into another builder between the parser and request.  They have made other areas easier to work with.

Reply:
Alright, made some updates to this on the [latest commit](https://github.com/saralein/java-server/pull/6/commits/bdbd93e446780008b4c8fd5a9f2e3699618a3238).